examples/pytorch/rgcn/edge_classify.py

```diff
@@ -0,0 +1,490 @@
+"""
+Relational GCN for edge classification
+
+Code follows the dgl tutorial:
+https://docs.dgl.ai/guide/minibatch-edge.html#guide-minibatch-edge-classification-sampler
+
+Data is generated by fake_data_generator.py
+
+Graph: 2 types of nodes (seller and product). A seller-product pair is an offer listing, ie. edge. Task is to predict
+whether an offer listing (edge) is defective.
+
+Model: use RGCN propagation to get node embeddings, and then concatenate node embeddings with edge embeddings to make a
+binary classification. The edge embeddings are generated from a neural network. All parameters are optimized as a whole
+by the cross entropy objective function.
+
+This code can handle:
+1. Features on different node type have different dimensions.
+2. Mini-batch training
+3. RGCN homogeneous implementation.
+4. Model performance is evaluated by accuracy and roc auc.
+
+"""
+import argparse
+import numpy as np
+import pandas as pd
+import time
+import torch
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl
+from dgl.nn.pytorch import RelGraphConv
+from sklearn.metrics import roc_auc_score
+
+from model import RelGraphEmbedLayer
+from fake_data_generator import load_generated_data
+
+class RGCN(nn.Module):
```

Comment:
Can we re-use the module from the node classification example?

Reply:
Done. Compared to your RGCN module in node classification example, I have a new arg "input feat_dim". Its dim is different from h_dim. In your example, your input feat dim == h_him.